lib/_http_client.js

```diff
@@ -140,13 +151,31 @@ function ClientRequest(options, cb) {
     self._last = true;
     self.shouldKeepAlive = false;
     if (options.createConnection) {
-      self.onSocket(options.createConnection(options));
+      options.oncreate = oncreate;
+      var newSocket = options.createConnection(options);
```

Comment:
That was what I tried initially but there was a bunch of problems because of some assumptions made elsewhere in the codebase. I cannot remember offhand, maybe it was related to `Https.Agent` having a different function signature for its `createConnection()`. I will have to look again to see what it was.